lnbits/tasks.py

```diff
@@ -181,3 +188,41 @@ async def invoice_callback_dispatcher(checking_id: str):
         for chan_name, send_chan in invoice_listeners.items():
             logger.trace(f"sse sending to chan: {chan_name}")
             await send_chan.put(payment)
+
+
+push_notification_key_file = os.path.join(
+    settings.lnbits_data_folder, "push_notification_vapid.{}"
+)
+
+
+def create_push_notification_keypair():
+    """
+    Generates VAPID key pair if not already exist
+    """
+    privkey = Vapid().from_file(push_notification_key_file.format("pem"))
+
+    if not os.path.exists(push_notification_key_file.format("pub")):
+        pubkey = privkey.public_key.public_bytes(
+            serialization.Encoding.X962, serialization.PublicFormat.UncompressedPoint
+        )
+        f = open(push_notification_key_file.format("pub"), "w")
+        f.write(b64urlencode(pubkey))
+        logger.info("✔️ Generated VAPID key pair for push notifications.")
+
+
+async def send_push_notification(subscription, title, body, url=""):
+    try:
+        logger.debug(f"sending push notification")
+        webpush(
+            json.loads(subscription.data),
+            json.dumps({"title": title, "body": body, "url": url}),
+            Vapid().from_file(push_notification_key_file.format("pem")),
+            {"aud": "", "sub": "mailto:devnull@example.com"},
```

Comment:
why is this email link needed? is that required for push notification? and why example?

Reply:
Explained it here: https://github.com/lnbits/lnbits/pull/1054#issuecomment-1355900490

Yes, it's mandatory by spec and you get error otherwise.